datafusion/physical-expr/src/expressions/cast.rs

```diff
@@ -167,6 +168,11 @@ impl PhysicalExpr for CastExpr {
     fn get_ordering(&self, children: &[SortProperties]) -> SortProperties {
         children[0]
```

Comment:
I wonder if a cast really preserves the ordering? I would have thought that casting numbers to strings would change the ordering for example

Reply:
Actually you are right, we should add a check something like whether cast is between similar types